tests/rustdoc/redirect.rs

```diff
@@ -9,8 +9,11 @@ pub trait Foo {}
 // @has redirect/index.html
 // @has - '//code' 'pub use reexp_stripped::Bar'
 // @has - '//code/a' 'Bar'
-// @has reexp_stripped/hidden/struct.Bar.html
-// @has - '//p/a' '../../reexp_stripped/struct.Bar.html'
+// `Bar` is in a `doc(hidden)` module so the link should point to a non-existing file.
+// However, the struct is reexported at the root of the crate so we should point to it.
+// @has - '//a[@href="../reexp_stripped/struct.Bar.html"]' 'Bar'
```

Comment:
Currently this test is failing. I need to find out why we generate a link to the original location of the item and not to the reexport directly.